kernel/sched.c

```diff
@@ -321,7 +321,7 @@ static inline bool is_aborting(struct k_thread *thread)
 }
 
 /* Return true if the thread is aborting or suspending, else false */
-static inline bool is_halting(struct k_thread *thread)
+static inline bool is_halting(volatile struct k_thread *thread)
```

Comment:
I noticed during debugging that on some architectures, the is_halting() loop in z_thread_abort() was degenerating into a tight loop of testing the register without reloading it between tests. This would be a problem even outside the corner case that I am trying to fix.

Reply:
I noticed that when building for qemu_x86_64, but I will check other boards as well.